### PR TITLE
GitHub action improvements

### PR DESCRIPTION
**Pull request by v-fearam**
1- Secret added by script using GitHub CLI
2- Create fork by script using GitHub CLI
3- Adding images to ACR in github workflow
4- Checking cluster creation by script

### CODE REVIEW

File: 06-aks-cluster.md [hunk 1]
```diff
@@ -120,7 +108,42 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
...
 
-        > :warning: You can continue only after the GitHub Workflow completes successfully
+    1.  You can continue only after the GitHub Workflow completes successfully
+
+        ```bash
+        # Region 1 - Wait until updated with provisioningState at 'Succeeded'.
+        az deployment group wait -n cluster-stamp -g rg-bu0001a0042-03 --updated
+        # Region 2- Wait until updated with provisioningState at 'Succeeded'.
```
Comment (ferantivero) [inline 06-aks-cluster.md:116]: have we considered `gh api /repos/:owner/:repo/actions/workflows` instead? 👀

File: 01-prerequisites.md [hunk 2]
```diff
@@ -22,15 +22,22 @@ This is the starting point for the instructions on deploying the [AKS Secure Bas
...
+   ```
 
-   > :twisted_rightwards_arrows: ensure references to this git repository mentioned throughout the walkthrough are updated to use your own fork.
+1. Fork the repository first, and clone it
 
    ```bash
+   gh repo fork mspnp/aks-secure-baseline --clone=true --remote=false
+   cd aks-secure-baseline
+   git remote remove upstream
    GITHUB_USER_NAME=<your username>
```
Comment (ferantivero) [inline 01-prerequisites.md:39]:
```suggestion
   GITHUB_USER_NAME=$(echo $(gh auth status 2>&1) | sed 's/.*as \(.*\)(.*/\1/')
```

File: 06-aks-cluster.md [hunk 3]
```diff
@@ -40,37 +40,25 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
...
 
-    1.  Create `AZURE_CREDENTIALS` secret in your GitHub repository. For more
-        information, please take a look at [Creating encrypted secrets for a repository](https://docs.github.com/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository).
+    1.  Create `AZURE_CREDENTIALS` secret in your GitHub repository.
 
         > :bulb: Use the content from the `sp.json` file.
 
         ```bash
-        cat sp.json
+        gh secret set AZURE_CREDENTIALS  -b"$(cat sp.json)" -repo="$GITHUB_USER_NAME/aks-secure-baseline"
```
Comment (ferantivero) [inline 06-aks-cluster.md:48]:
```suggestion
        gh secret set AZURE_CREDENTIALS  -b"$(cat sp.json)" -repo=":owner/:repo"
```

File: 06-aks-cluster.md [hunk 4]
```diff
@@ -40,37 +40,25 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
...
-        > :bulb:
-        >
-        > - Use the env var value of `APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64` and `APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64`
-        > - Ideally fetching this secret from a platform-managed secret store such as [Azure KeyVault](https://github.com/marketplace/actions/azure-key-vault-get-secrets)
+    1.  Create `APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64` and `APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64` secret in your GitHub repository.
 
         ```bash
-        echo $APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64
-        echo $APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64
+        gh secret set APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64  -b"${APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64}" -repo="$GITHUB_USER_NAME/aks-secure-baseline"
```
Comment (ferantivero) [inline 06-aks-cluster.md:54]:
```suggestion
        gh secret set APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64  -b"${APP_GATEWAY_LISTENER_REGION1_CERTIFICATE_BASE64}" -repo=":owner/:repo"
```

File: 06-aks-cluster.md [hunk 5]
```diff
@@ -40,37 +40,25 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
...
+        gh secret set APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64  -b"${APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64}" -repo="$GITHUB_USER_NAME/aks-secure-baseline"
```
Comment (ferantivero) [inline 06-aks-cluster.md:55]:
```suggestion
        gh secret set APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64  -b"${APP_GATEWAY_LISTENER_REGION2_CERTIFICATE_BASE64}" -repo=":owner/:repo"
```

File: 06-aks-cluster.md [hunk 6]
```diff
@@ -40,37 +40,25 @@ Now that the [cluster prequisites and shared Azure service instances are provisi
...
-
-        > :bulb:
-        >
-        > - Use the env var value of `AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64`
-        > - Ideally fetching this secret from a platform-managed secret store such as [Azure Key Vault](https://github.com/marketplace/actions/azure-key-vault-get-secrets)
+    1.  Create `AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64` secret in your GitHub repository.
 
         ```bash
-        echo $AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64
+        gh secret set AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64  -b"${AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64}" -repo="$GITHUB_USER_NAME/aks-secure-baseline"
```
Comment (ferantivero) [inline 06-aks-cluster.md:61]:
```suggestion
        gh secret set AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64  -b"${AKS_INGRESS_CONTROLLER_CERTIFICATE_BASE64}" -repo=":owner/:repo"
```

File: 10-validation.md [hunk 7]
```diff
@@ -20,7 +20,7 @@ This section will help you to validate the workload is exposed correctly and res
 1. Browse to the site.
 
    ```bash
-   echo $FRONTDOOR_FQDN
+   echo https://$FRONTDOOR_FQDN
```
Comment (ferantivero) [inline 10-validation.md:23]: please let's revert this change to avoid conflicts. I'm already taking care of this section from another PR.

**Review by ferantivero — CHANGES_REQUESTED** [review]

File: 10-validation.md (hunk 7 above)
Comment (v-fearam) [reply 10-validation.md]: done

File: 06-aks-cluster.md (hunk 6 above)
Comment (v-fearam) [reply 06-aks-cluster.md]: done

File: 06-aks-cluster.md (hunk 5 above)
Comment (v-fearam) [reply 06-aks-cluster.md]: done

File: 06-aks-cluster.md (hunk 4 above)
Comment (v-fearam) [reply 06-aks-cluster.md]: done

File: 06-aks-cluster.md (hunk 3 above)
Comment (v-fearam) [reply 06-aks-cluster.md]: done

File: 01-prerequisites.md (hunk 2 above)
Comment (v-fearam) [reply 01-prerequisites.md]: done

File: 01-prerequisites.md [hunk 8]
```diff
@@ -22,15 +22,22 @@ This is the starting point for the instructions on deploying the [AKS Secure Bas
 
    [![Launch Azure Cloud Shell](https://docs.microsoft.com/azure/includes/media/cloud-shell-try-it/launchcloudshell.png)](https://shell.azure.com)
 
-1. Fork the repository first, and clone it
+1. Ensure [GitHub CLI installed](https://github.com/cli/cli)
```
[the remark this answers is not in the document]
Comment (v-fearam) [reply 01-prerequisites.md]: done

File: 06-aks-cluster.md (hunk 1 above)
Comment (v-fearam) [reply 06-aks-cluster.md:116]: Response:
```
{
  "total_count": 1,
  "workflows": [
    {
      "id": 6532436,
      "node_id": "MDg6V29ya2Zsb3c2NTMyNDM2",
      "name": "Deploy AKS Secure Baseline clusters stamp and Flux in different regions",
      "path": ".github/workflows/aks-deploy.yaml",
      "state": "active",
      "created_at": "2021-03-09T09:00:30.000-03:00",
      "updated_at": "2021-03-09T09:00:30.000-03:00",
      "url": "https://api.github.com/repos/v-fearam/aks-secure-baseline/actions/workflows/6532436",
      "html_url": "https://github.com/v-fearam/aks-secure-baseline/blob/main/.github/workflows/aks-deploy.yaml",
      "badge_url": "https://github.com/v-fearam/aks-secure-baseline/workflows/Deploy%20AKS%20Secure%20Baseline%20clusters%20stamp%20and%20Flux%20in%20different%20regions/badge.svg"
    }
  ]
}
```
This is information about workflows defined, not executions states

Comment (v-fearam) [reply 06-aks-cluster.md]: Thank for the support Fernando 
until export GH_WF_STATUS=$(gh api /repos/:owner/:repo/actions/runs/$(gh api /repos/:owner/:repo/actions/runs -q ".workflow_runs[0].id") -q ".status" 2> /dev/null) && [[ $GH_WF_STATUS == "completed" ]]; do echo "Monitoring GitHub workflow execution: ${GH_WF_STATUS}" && sleep 20; done

**Comment by ferantivero** [comment]
LGTM 🚀
